tests/benchdnn/graph/graph_memory.cpp

```diff
@@ -126,8 +131,8 @@ int dnn_graph_mem_t::fill_mem_with_data(const dnn_mem_t &mem) {
         std::memcpy(graph_data_handle, prim_data_handle, graph_mem.size());
     };
 
-    if (src_dt != dst_dt) {
-        dnn_mem_t c_mem(ndims, mem.dims(), dst_dt, mtag, g_eng.get());
+    if (src_dt != dst_dt || src_eng != dst_eng) {
+        dnn_mem_t c_mem(ndims, mem.dims(), dst_dt, mtag, dst_eng);
```

Comment:
A comment a-la "If `eng` is different from `mem_eng` need to transfer data under same engine to perform a data copy." would be helpful.